Refactor number emitting (match precision of ruby sass)

mgreter opened the pull request:
Fixes https://github.com/sass/libsass/issues/1153

xzyfer commented on inspect.cpp:
```diff
@@ -416,43 +416,91 @@ namespace Sass {
+    while (*it && *it == '0') ++ it;
+    bool is_int = (*it == 0);
+    // take a shortcut for integers
+    if (is_int)
```
`bool is_int = d.find('.') == string::npos` ?

xzyfer commented on inspect.cpp:
```diff
@@ -416,43 +416,91 @@ namespace Sass {
+    }
+
+    const char *it = ss.str().c_str();
+    while (*it && *it != '.' && *it != ',') ++ it;
```
I find this code really unintuitive and hard to grok. We should aim for expressive code first, and optimize later.

```
bool has_decimal = ss.find_first_of(".,");
```

mgreter replied:
Unoftunately every bit (even the number 12) seems to be exactly needed this way.  But I guess this one loop can be rewritten with find_first_of ...

xzyfer commented on inspect.cpp:
```diff
@@ -416,43 +416,91 @@ namespace Sass {
-    if (ctx->precision == 0) d += ".0";
-    // append number and unit
-    append_token(d + n->unit(), n);
+      error(res + " isn't a valid CSS value.", n->pstate());
```
Should this be in the output phase? Surely we know if a number in valid in eval?

mgreter replied:
A number can eval to something that is not valid, like:

```
$a: 2px * 2px; // ->invalid
div { margin: $a / 2px; } // ->valid
```

mgreter commented:
Refactored the code and incorporated the feedback from @xzyfer. Made the code much more explicit. I think I cannot really improve much on it anymore. Hope CI build passes this times. IMHO appveyor sometimes fails for one build (with random errors it seems) ...

xzyfer commented:
:+1:
